test/object-store/sync/app.cpp

```diff
@@ -3053,6 +3062,40 @@ TEST_CASE("app: sync integration", "[sync][app]") {
         }
     }
 
+    SECTION("pausing a session does not hold the DB open") {
+        SyncTestFile config(app, partition, schema);
+        DBRef dbref;
+        std::shared_ptr<SyncSession> sync_sess_ext_ref;
+        {
+            auto realm = Realm::get_shared_realm(config);
+            wait_for_download(*realm);
+
+            auto state = realm->sync_session()->state();
+            REQUIRE(state == SyncSession::State::Active);
+
+            sync_sess_ext_ref = realm->sync_session()->external_reference();
+            dbref = TestHelper::get_db(*realm);
+            REQUIRE(dbref.use_count() == 4);
+
+            realm->sync_session()->pause();
+            state = realm->sync_session()->state();
+            REQUIRE(state == SyncSession::State::Paused);
+        }
+
+        REQUIRE_THAT(
+            [&] {
+                return dbref.use_count() == 2;
+            },
+            ReturnsTrueWithinTimeLimit{});
+
+        sync_sess_ext_ref.reset();
+        REQUIRE_THAT(
+            [&] {
+                return dbref.use_count() == 1;
+            },
+            ReturnsTrueWithinTimeLimit{});
+    }
```

Comment:
Can we also check that `resume()` works as intended?

Reply:
We have tests that resume() resumes the session, what else would you like to see tested here?